### PR TITLE
DRILL-6612: Query fails with AssertionError when joining persistent a…

### PR DESCRIPTION
**Pull request by vvysotskyi**
…nd temporary tables

For details please see [DRILL-6612](https://issues.apache.org/jira/browse/DRILL-6612)

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SchemaUtilites.java [hunk 1]
```diff
@@ -144,6 +144,11 @@ public static String getSchemaPath(List<String> schemaPath) {
     return SCHEMA_PATH_JOINER.join(schemaPath);
   }
 
+  /** Utility method to get the list with schema path components for given schema path string. */
+  public static List<String> getSchemaPathList(String schemaPath) {
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SchemaUtilites.java]: Please rename to `getSchemaPathAsList`.

Comment (vvysotskyi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SchemaUtilites.java]: Thanks, done.

**Comment by arina-ielchiieva** [comment]
+1, LGTM.

**Comment by vvysotskyi** [comment]
@arina-ielchiieva, thanks for the review!

**Pull request merged**